packages/gatsby-telemetry/src/event-storage.ts

```diff
@@ -1,34 +1,44 @@
-const os = require(`os`)
-const path = require(`path`)
-const Store = require(`./store`)
-const fetch = require(`node-fetch`)
-const Configstore = require(`configstore`)
-const { ensureDirSync } = require(`fs-extra`)
-
+import os from "os"
+import path from "path"
+import Configstore from "configstore"
+import createFetch from "@turist/fetch"
+import { Store } from "./store"
+import { ensureDirSync } from "fs-extra"
 import { isTruthy } from "./is-truthy"
 
+const fetch = createFetch()
+
 /* The events data collection is a spooled process that
  * buffers events to a local fs based buffer
  * which then is asynchronously flushed to the server.
- * This both increases the fault tolerancy and allows collection
+ * This both increases the fault tolerance and allows collection
  * to continue even when working offline.
  */
 module.exports = class EventStorage {
   analyticsApi =
     process.env.GATSBY_TELEMETRY_API || `https://analytics.gatsbyjs.com/events`
+  // eslint-disable-line @typescript-eslint/no-explicit-any
+  config: Configstore | Record<string, any>
+  store: Store
+  verbose: boolean
+  debugEvents: boolean
+  disabled: boolean
+
   constructor() {
     try {
       this.config = new Configstore(`gatsby`, {}, { globalConfigPath: true })
     } catch (e) {
       // This should never happen
```

Comment:
If this should never throw, why are we going to all this effort to manually build the store? Can't we just bail? If not, it would be better to create a separate FallbackConfigstore class an instantiate that.

Reply:
It will throw when $HOME is not writable. that actually happens a lot :\

Reply:
and we can have this thing work with an in-memory store in those cases, tho maybe the uuid should be a bit more distinct

Reply:
I'd remove that comment and replace it with that explanation. I do think having a class for `InMemoryConfigStore` would be better than constructing bit-by-bit, old-skool like here.

Reply:
yea, it would be a good refactoring thing to do over this quick workaround, I'd like to leave that to another PR

Reply:
It would be nice to get that in this PR if possible. It shouldn;t be a big change: just move the stuff you've already written outside of this class into a class of its own.

Reply:
done